tools/clang/unittests/HLSL/PixTestUtils.cpp

```diff
@@ -274,6 +274,57 @@ PassOutput RunAnnotationPasses(dxc::DxcDllSupport &dllSupport, IDxcBlob *dxil,
           Tokenize(outputText.c_str(), "\n")};
 }
 
+class InstructionOffsetSeekerImpl : public InstructionOffsetSeeker {
+
+  std::map<std::wstring, DWORD> m_labelToInstructionOffset;
+
+public:
+  InstructionOffsetSeekerImpl(DebuggerInterfaces &debuggerInterfaces) {
+    DWORD SourceFileOrdinal = 0;
+    CComBSTR fileName;
+    CComBSTR fileContent;
+    while (SUCCEEDED(debuggerInterfaces.compilationInfo->GetSourceFile(
+        SourceFileOrdinal, &fileName, &fileContent))) {
+      auto lines = strtok(std::wstring(fileContent), L"\n");
+      for (size_t line = 0; line < lines.size(); ++line) {
+        const wchar_t DebugLocLabel[] = L"debug-loc(";
+        auto DebugLocPos = lines[line].find(DebugLocLabel);
+        if (DebugLocPos != std::wstring::npos) {
+          auto StartLabelPos = DebugLocPos + (_countof(DebugLocLabel) - 1);
+          auto CloseDebugLocPos = lines[line].find(L")", StartLabelPos);
+          VERIFY_ARE_NOT_EQUAL(CloseDebugLocPos, std::string::npos);
```

Comment:
I wonder about the utility of using VERIFY macros deep in test utility code, especially within loops and such.  These should only fail on a malformed test or an internal test bug.  We certainly do use it for some things of this class (probably too much), but these add to the verification noise, since it's not what the test case is focused on testing.  Logging a failure with some detail is important, but when using VERIFY macros, every passing verification is logged for use if something fails later in the test, or if verbose logging is enabled.

Reply:
Sure. It was pretty noisy.